packages/ui/primitives/react-hooks/src/useAsyncCallback.ts

```diff
@@ -4,6 +4,10 @@
 
 import { useEffect, useState } from 'react';
 
+/**
+ * @deprecated
+ */
+// TODO(burdon): Clashes with useAsyncEffect.
```

Comment:
`useAsyncEffect` is the one that should stop being used. While I've never used this, it looks fine to me.